### ISSUE TITLE
End of the day sales form

### ISSUE THREAD

**Issue by tannakartikey** [ajc113/inventory#7]
Sure, we can approach this in a step-by-step process. Here, I'll provide you with a conceptual outline rather than specific code, as the exact implementation will vary depending on the language and technologies you're using (e.g., Rails, Django, Node.js, Laravel, etc). 

### 1. Dynamic Sales Form

The form should be dynamically generated based on the `location_id`. The list of available flavors and their quantity will be fetched from a database table that maintains a record of all the flavors available at each location.

The form will look something like this:

| Flavor | Quantity |
|---|---|
| Vanilla | [0] |
| Chocolate | [0] |

In the brackets, we have the input fields that take only numbers. The default value will be zero.

### 2. Form Submission and Quantity Update

Once the form is submitted, the server will receive the form data and will start processing it. For each flavor in the form, it will find the flavor record in the database associated with the current `location_id` and subtract the submitted quantity from the existing quantity. 

### 3. Sales Table

In parallel, the server will also create a new entry in the `sales` table. This table will have a `belongs_to` relationship with both the `flavor` and `location` tables. The `sales` table might look something like this:

| id | flavor_id | location_id | quantity | created_at | updated_at |
|---|---|---|---|---|---|
| 1 | 2 | 3 | 3 | 2023-07-19 11:00:00 | 2023-07-19 11:00:00 |

Where `flavor_id` corresponds to the flavor sold (e.g., vanilla), `location_id` corresponds to the location where the sale occurred (e.g., Kingston), `quantity` is the number of units sold, and the timestamps provide the timeline.

### 4. Similar Process for Manufacturing

A similar approach will be taken for the manufacturing process, but the details of that will be covered in a separate ticket.
